mistralrs-core/src/sequence.rs

```diff
@@ -244,7 +267,14 @@ impl Sequence {
         } else if self.tokens.len().saturating_sub(self.prompt_len) == max_model_len {
             Some(StopReason::ModelLength(max_model_len))
         } else {
-            None
+            let text_acc = self.generated_text(tokenizer);
+            if self.stop_strings.iter().any(|s| text_acc.contains(s)) {
+                // TODO: a different stop reason?
+                // TODO: strip the stop string from the generated text?
+                Some(StopReason::StopTok(tok))
```

Comment:
StopReason implements Copy, so I could not create a variant for the String. I need to think more about this, maybe it just shouldn't be copy.

Reply:
@EricLBuehler what do you think? Removing Copy raised a lot of errors...

Reply:
I made it save the index of the string in the list of stop strings. Now it's copy too.